Firestore/core/src/firebase/firestore/model/mutation.h

```diff
@@ -178,18 +186,28 @@ class Mutation {
       const MaybeDocument* base_doc,
       const Timestamp& local_write_time) const = 0;
 
+  friend bool operator==(const Mutation& lhs, const Mutation& rhs) {
```

Comment:
optional: I feel moving the inline below close to operator!= and keep here just a friend declaration is easier to read. I cannot find any style guide on this and thus optional.